backend/src/api/difficulty-adjustment.ts

```diff
@@ -41,14 +45,14 @@ class DifficultyAdjustmentApi {
       if (timeAvgMins > 20) {
         timeAvgMins = 20;
       }
-      if (now - latestBlock.timestamp + timeAvgMins * 60 > 1200) {
-        timeOffset = -Math.min(now - latestBlock.timestamp, 1200) * 1000;
```

Comment:
Q: What is this for? It is set to the milliseconds value of time since last block timestamp with a hard cap at 20 minutes (in ms) multiplied by -1 And it is returned as-is... What is its usecase? If not testnet, it just is ignored and stays 0.

Reply:
I added a test case that barely triggers this, and it makes the data's meaning a bit more clear... however, I am unsure what the use case is, still.

Reply:
According to @antonilol who made this and wrote the code comment above, this code is made for _testnet_ only since this code is wrapped in this.

So as far as I know it works and shouldn't be deleted

```
/ Testnet difficulty is set to 1 after 20 minutes of no blocks,
// therefore the time between blocks will always be below 20 minutes (1200s).
if (config.MEMPOOL.NETWORK === 'testnet') {
```

Reply:
a negative timeOffset lowers the estimated mine time of a mempool block (the `In ~n minutes`) to ensure the estimation never shows a gap between blocks of more than 20 minutes. on all other networks this value is zero so zero is added to a number = no effect

it is used here: https://github.com/mempool/mempool/blob/master/frontend/src/app/components/mempool-blocks/mempool-blocks.component.html#L29

this rule in the bitcoin source code: https://github.com/bitcoin/bitcoin/blob/master/src/pow.cpp#L23-L27